### PR TITLE
add GetCellRichText method and test

### PR DESCRIPTION
**Pull request by tonny-zhang**
# GetCellRichText 

## Description
You can get text and font of cell by use GetCellRichText

```go
        f := NewFile()

	runsSource := []RichTextRun{
		{
			Text: "a\n",
		},
		{
			Text: "b",
			Font: &Font{
				Underline: "single",
				Color:     "ff0000",
				Bold:      true,
				Italic:    true,
				Family:    "宋体",
				Size:      100,
				Strike:    true,
			},
		},
	}
	f.SetCellRichText("Sheet1", "A1", runsSource)

	runs, err := f.GetCellRichText("Sheet1", "A1")
```
## Related Issue

https://github.com/360EntSecGroup-Skylar/excelize/issues/696

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: cell.go [hunk 1]
```diff
@@ -494,6 +494,57 @@ func (f *File) SetCellHyperLink(sheet, axis, link, linkType string) error {
...
+
+	sst := f.sharedStringsReader()
+	a := sst.SI[siIndex]
+	for _, v := range a.R {
+		run := RichTextRun{
+			Text: v.T.Val,
+		}
+		if nil != v.RPr {
+			font := Font{}
+			font.Bold = v.RPr.B == " "
```
Comment (xuri) [inline cell.go:522]: Different with set the rich text, by OOXML spec, we need also handle italic and bold style created by Excel and other application, which maybe will not have an empty string, and just have a self closed XML element such as `</b>`, `</i>` or `</strike>`, etc, I purpose to change the data type of following fields in the `xlsxRPr` structure to a pointer of string like this:

```go
B         *string         `xml:"b"`
I         *string         `xml:"i"`
Strike    *string         `xml:"strike"`
Outline   *string         `xml:"outline"`
Shadow    *string         `xml:"shadow"`
Condense  *string         `xml:"condense"`
Extend    *string         `xml:"extend"` 
```

you know this change will affect the set rich text.

**Review by xuri — CHANGES_REQUESTED** [review]
Thanks for your pull request, I have left some comments and I'll certainly accept this feature if it is done.

**Comment by xuri** [comment]
Hi @tonny-zhang, I have fixed it base on your codes and I'll merge this PR after the CI check pass.

**Comment by tonny-zhang** [comment]
[GetCellRichText.xlsx](https://github.com/360EntSecGroup-Skylar/excelize/files/6022252/GetCellRichText.xlsx)

```
	t.Run("init from file", func(t *testing.T) {
		f, _ := OpenFile(path.Join("test", "GetCellRichText.xlsx"))
		runs, e := f.GetCellRichText("Sheet1", "C1")
	})
```

this test case can't get error underline and get panic

**Comment by tonny-zhang** [comment]
[GetCellRichText.xlsx](https://github.com/360EntSecGroup-Skylar/excelize/files/6022252/GetCellRichText.xlsx)

```
	t.Run("init from file", func(t *testing.T) {
		f, _ := OpenFile(path.Join("test", "GetCellRichText.xlsx"))
		runs, e := f.GetCellRichText("Sheet1", "C1")
	})
```

this test case can't get error underline and get panic

**Comment by tonny-zhang** [comment]
[GetCellRichText.xlsx](https://github.com/360EntSecGroup-Skylar/excelize/files/6022252/GetCellRichText.xlsx)

```
	t.Run("init from file", func(t *testing.T) {
		f, _ := OpenFile(path.Join("test", "GetCellRichText.xlsx"))
		runs, e := f.GetCellRichText("Sheet1", "C1")
	})
```

this test case can't get right underline and get panic

File: cell.go [hunk 2]
```diff
@@ -505,41 +505,38 @@ func (f *File) GetCellRichText(sheet, cell string) (runs []RichTextRun, err erro
...
 			Text: v.T.Val,
 		}
 		if nil != v.RPr {
 			font := Font{}
-			font.Bold = v.RPr.B == " "
-			font.Italic = v.RPr.I == " "
+			font.Bold = v.RPr.B != nil
+			font.Italic = v.RPr.I != nil
 			if nil != v.RPr.U {
 				font.Underline = *v.RPr.U.Val
```
Comment (tonny-zhang) [inline cell.go:526]: underline has two formats `<u/>` and `<u val="single"></u>`, so it should check `v.RPr.U.Val`
```
if nil != v.RPr.U {
	if nil != v.RPr.U.Val {
		font.Underline = *v.RPr.U.Val
	} else {
		// use single for default
		font.Underline = "single"
	}
}
```

**Review by tonny-zhang — COMMENTED** [review]
I have left some comments

**Comment by xuri** [comment]
> [GetCellRichText.xlsx](https://github.com/360EntSecGroup-Skylar/excelize/files/6022252/GetCellRichText.xlsx)
> 
> ```
> 	t.Run("init from file", func(t *testing.T) {
> 		f, _ := OpenFile(path.Join("test", "GetCellRichText.xlsx"))
> 		runs, e := f.GetCellRichText("Sheet1", "C1")
> 	})
> ```
> 
> this test case can't get right underline and get panic

Thanks for your feedback, I have fixed on commit [`d840509`](https://github.com/360EntSecGroup-Skylar/excelize/commit/d84050921eddf5c4221f8723477be2ac93f56ecc#diff-d74b3934a69be62b4f188d57aaf2dabd9c2375cd91f419029d266e630c3c9792L525)

**Comment by tonny-zhang** [comment]
@xuri I don't think [commit]( https://github.com/360EntSecGroup-Skylar/excelize/commit/d84050921eddf5c4221f8723477be2ac93f56ecc#diff-d74b3934a69be62b4f188d57aaf2dabd9c2375cd91f419029d266e630c3c9792L525) can get underline value for `<u/>`, should give a default value ?

```go
if nil != v.RPr.U {
	if nil != v.RPr.U.Val {
		font.Underline = *v.RPr.U.Val
	} else {
		// use single as default
		font.Underline = "single"
	}
}
```

**Comment by xuri** [comment]
Yep, and I purpose to set the default value of underline as `none`:

```go
font := Font{ Underline: "none"}

...

if nil != v.RPr.U {
    font.Underline = "single"
    if nil != v.RPr.U.Val {
        font.Underline = *v.RPr.U.Val
    }
}
```

**Pull request merged**